install_with_docker.sh

```diff
@@ -180,27 +180,49 @@ https://api.segment.io/v1/track \
   }
 }'
 
-echo ""
-tput bold
-echo -e "\n===> ${bold}Run the following command to start the siglens server"
-tput sgr0
-echo "====================**************************************************============================"
-echo ""
-echo -e "docker run -it --mount type=bind,source="$(pwd)"/data,target=/siglens/data \
-    --mount type=bind,source="$(pwd)"/server.yaml,target=/siglens/server.yaml \
-    -p 8081:8081 -p 80:80 siglens/siglens:0.1.0"
-echo ""
-echo "====================**************************************************============================"
+initial_port=5122
+start_port=5122
+end_port=5122
+
+check_ports() {
+    
+    if lsof -Pi :$initial_port -sTCP:LISTEN -t > /dev/null || docker ps --format "{{.Ports}}" | grep -q "0.0.0.0:${initial_port}->"; then
+        for port in $(seq $start_port $end_port); do
+            if lsof -Pi :$port -sTCP:LISTEN -t > /dev/null || docker ps --format "{{.Ports}}" | grep -q "0.0.0.0:$port->"; then
+                continue
+            else
+                echo $port
+                return 0
+            fi
+        done
+        echo "-1"
+        return 1
+    else
+        echo "${initial_port}"
+        return 0
+    fi
+}
+
+ui_port=$(check_ports)
+
+if [ ${ui_port} == "-1" ]; then
+    echo ""
+    tput bold
+    printf "\e[31mError: No available port found on port ${initial_port} or in the range from ${start_port} to ${end_port}.\e[0m\n"
```

Comment:
Change to "no available port found , checked range [startport -- endport]" ... something like that